File: 백준/Bronze/1157. 단어 공부/단어 공부.py
inputText = open(0).readline().strip().upper()
abcDict = dict()
for letter in inputText:
    if letter in abcDict:
        abcDict[letter] += 1
    else:
        abcDict[letter] = 1

abcList = list(abcDict.keys())
countList = list(abcDict.values())

maxIndex = countList.index(max(countList))
maxLetter = abcList[maxIndex]

countList.sort(reverse=True)
try:
    if countList[0] == countList[1]:
        maxLetter = "?"
except:
    pass
print(maxLetter)

File: 백준/Bronze/1157. 단어 공부/단어 공부.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        char[] inputText = br.readLine().strip().toCharArray();
        int[] counter = new int[26];

        for (char c : inputText) {
            int n;
            if (c >= 'a') {
                n = c - 'a';
            } else {
                n = c - 'A';
            }
            counter[n]++;
        }
        int max = 0;
        int maxIdx = 0;
        boolean unknown = false;
        for (int i=0;i<26; i++){
            if (counter[i]>max){
                max = counter[i];
                maxIdx = i;
                unknown = false;
            } else if (counter[i]==max){
                unknown = true;
            }
        }
        System.out.println(unknown?'?':(char)(maxIdx+'A'));
    }
}
